File: DSA/Arshdeep-Kaur/Day 43/Day43_Question2.cpp
class Solution {
public:
    int numTimesAllBlue(vector<int>& flips) {
        int moves=0;

        int right = 0; 

        for(int i=0; i<flips.size(); i++){
            right = max(right, flips[i]);
            if(right == i+1){
                moves++;
            }
        }
        return moves;
    }
};

File: DSA/Manav-Rajput/Day43_question2.java
class Solution {
    public int numTimesAllBlue(int[] flips) {
        int n = flips.length;
        int maxIndex = 0; 
        int prefixAlignCount = 0; 
        
        for (int i = 0; i < n; i++) {
            maxIndex = Math.max(maxIndex, flips[i]);
            if (maxIndex == i + 1) {
                prefixAlignCount++;
            }
        }
         return prefixAlignCount;
    }
}
